### PR TITLE
doc: add labels documentation.

### PR DESCRIPTION
**Pull request by ldez**
### Description

Add an explanation for each label we use.

https://mensuel.framapad.org/p/EYZcWzVbgO

### CODE REVIEW

File: .github/MAINTAINER.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+area/api:: Traefik API related.
+area/logs:: Traefik logs related.
+area/sticky-session:: Sticky session related.
+area/websocket:: WebSocket related.
+area/webui:: Web UI related.
+area/infrastructure:: related to CI or Traefik building scripts.
+area/documentation:: regards improving/adding documentation
+
+priority/P0:: needs hot fix. (only for issue)
+priority/P1:: fixed in next release. (only for issue)
```
Comment (timoreimann) [inline .github/MAINTAINER.adoc]: `fixed` sounds like this is only meant for bugs. Should we say `shipping` instead?

(Same for P2.)

File: .github/MAINTAINER.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+priority/P2:: fixed in the future. (only for issue)
+priority/P3:: maybe. (only for issue)
```
Comment (timoreimann) [inline .github/MAINTAINER.adoc]: s/maybe/backlog/?

File: .github/MAINTAINER.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+bug/possible:: (only for issues) if we need to analyze to understand if it's a bug or not. [bot only]
+bug/confirmed:: (only for issues) we are sure, it's a bug.
+bug/fix:: (only for PR) it's a bug fix.
+
+contributor/need-more-information:: we need more information from the contributor in order to analyze a problem.
+contributor/waiting-for-corrections:: (only for PR) we need the contributor to take actions in order to move forward with a PR
+contributor/needs-rebase:: (only for PR) use it only when there is some conflicts (and an automatic rebase is not possible) [bot, humans]
+
+duplicate:: it's a duplicate issue/PR.
```
Comment (timoreimann) [inline .github/MAINTAINER.adoc]: Let's move `duplicate` next to `declined`.

I also suggest to prefix it with with `resolution/`.

File: .github/MAINTAINER.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+enhancement:: a new or improved feature
```
Comment (timoreimann) [inline .github/MAINTAINER.adoc]: Can we prefix `enhancement`, `question`, and `proposal` with `kind/` (similar to how Docker and k8s do it)?

**Comment by ldez** [comment]
Asciidoc have better render for everything: table, list, image, video, ...

File: .github/MAINTAINER.adoc (hunk 1 above)
Comment (ldez) [inline .github/MAINTAINER.adoc]: Or `need to be fixed in ...`

**Comment by emilevauge** [comment]
@ldez 
>Asciidoc have better render for everything: table, list, image, video, ...

We all agree on this. But I think that contributors may be unwilling to use/learn Asciidoc instead of the simple/widespread Markdown. I think that the documentation should be the easiest possible to modify by contributors (even if Asciidoc is better and not that complex, Markdown is simpler).

File: .github/MAINTAINER.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+* Proposal issues are design proposal that need to be refined with multiple contributors.
+* Proposal PRs are technical prototypes that need to be refined with multiple contributors.
+
+declined:: Rule #1 of open-source: no is temporary, yes is forever.
+
+WIP:: (only for PR) Work In Progress.
+
+platform/windows:: Windows related.
+
+area/provider/boltdb:: Boltd DB related.
```
Comment (vdemeester) [inline .github/MAINTAINER.adoc]: We also should have a `area/provider` one (something that would affect *all* the providers for example)

**Comment by ldez** [comment]
You can use Markdown syntax in a Asciidoc document.

**Comment by emilevauge** [comment]
@ldez I don't think we want/need to change the documentation format right now. And we really don't want to have 2 format at the same time.

**Comment by timoreimann** [comment]
Right now, the `bug/*` labels are the only ones that stand out. I think we should also group them under the `kind/` namespace, e.g., `kind/bug/possible`, `kind/bug/confirmed`, and `kind/bug/fix`. This would be consistent IMO.

**Comment by ldez** [comment]
`kind/quality` it's still in discus for me.

**Pull request merged**